File: app/src/main/java/com/micewine/emu/viewmodels/ViewModelAppLogs.java
package com.micewine.emu.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ViewModelAppLogs extends ViewModel {
    private MutableLiveData<String> logsText = new MutableLiveData<>();

    public LiveData<String> getTextLiveData() {
        return logsText;
    }

    public void setText(String text) {
        logsText.setValue(text);
    }
}

File: app/src/main/java/com/micewine/emu/viewmodels/ViewModelAppLogs.kt
package com.micewine.emu.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class ViewModelAppLogs : ViewModel() {
    private val logsText = MutableLiveData<String>()
    val textLiveData: LiveData<String>
        get() = logsText

    fun setText(text: String) {
        logsText.value = text
    }
}